llvm/lib/AsmParser/LLParser.cpp

```diff
@@ -5824,6 +5839,17 @@ bool LLParser::convertValIDToValue(Type *Ty, ValID &ID, Value *&V,
                                "' but expected '" + getTypeString(Ty) + "'");
     V = ID.ConstantVal;
     return false;
+  case ValID::t_ConstantSplat:
+    if (!Ty->isVectorTy())
+      return error(ID.Loc, "vector constant must have vector type");
```

Comment:
If you feel like it, we could test these error conditions, but I'm not going to insist...

Reply:
Sure, I'll add them.  If more existed I would not have made my original mistake :)